indico/modules/events/editing/operations.py

```diff
@@ -283,3 +289,47 @@ def update_review_condition(condition, file_types):
 def delete_review_condition(condition):
     logger.info('Review condition %r deleted by %r', condition, session.user)
     db.session.delete(condition)
+
+
+def assign_editor(editable, editor):
+    editable.editor = editor
+    db.session.flush()
+    logger.info('Editable %r assigned to %r editor', editable, editor)
+
+
+def unassign_editor(editable):
+    editor = editable.editor
+    editable.editor = None
+    db.session.flush()
+    logger.info('Editor %r has been unassigned from %r', editor, editable)
+
+
+def generate_editables_zip(editables):
+    buf = BytesIO()
+    with ZipFile(buf, 'w', allowZip64=True) as zip_file:
+        for editable in editables:
+            for revision_file in editable.revisions[-1].files:
+                file_obj = revision_file.file
+                with file_obj.storage.get_local_path(file_obj.storage_file_id) as src_file:
+                    zip_file.write(src_file, _compose_filepath(editable, revision_file))
+
+    buf.seek(0)
+    return send_file('files.zip', buf, 'application/zip', inline=False)
+
+
+def _compose_filepath(editable, revision_file):
+    file_obj = revision_file.file
+    contrib = editable.contribution
+    editable_type = editable.type.name
+    code = 'Editable-{}'.format(editable.id)
```

Comment:
I don't think the internal ID makes sense here - I'd use the friendly id of the contribution instead.